k2/csrc/log.cuh

```diff
@@ -123,13 +127,68 @@ class Logger {
 
 class Voidifier {
  public:
-  __host__ __device__ void operator&(const Logger &)const {}
+  K2_CUDA_HOSTDEV void operator&(const Logger &) const {}
 };
 
+/**
+ * @brief
+ * Check if it's compiled in C++11 mode.
+ * It's codes from glog.
+ *
+ * @details
+ * GXX_EXPERIMENTAL_CXX0X is defined by gcc and clang up to at least
+ * gcc-4.7 and clang-3.1 (2011-12-13).  __cplusplus was defined to 1
+ * in gcc before 4.7 (Crosstool 16) and clang before 3.1, but is
+ * defined according to the language version in effect thereafter.
+ * Microsoft Visual Studio 14 (2015) sets __cplusplus==199711 despite
+ * reasonably good C++11 support, so we set LANG_CXX for it and
+ * newer versions (_MSC_VER >= 1900).
+ */
+#if (defined(__GXX_EXPERIMENTAL_CXX0X__) || __cplusplus >= 201103L || \
```

Comment:
I dont think we need this... too complicated.  Just K2_CHECK(ptr != NULL) is clearer and simpler.

Reply:
done